.github/workflows/checkmycode.yml

```diff
@@ -1,30 +1,165 @@
+    - uses: platisd/duplicate-code-detection-tool@master
+      with:
+        directories: "test/"
+        # Only examine .h and .cpp files
+        file_extensions: "sh, js"
+        # Only report similarities above 5%
+        ignore_below: 5
+        # If a file is more than 70% similar to another, then the job fails
+        fail_above: 70
+        # If a file is more than 15% similar to another, show a warning symbol in the report
+        warn_above: 15    
+  sonar_scan:
+    name: Detect bugs, vulnerabilities and code smells
+    needs: 
+      - duplicate_code_detection
+    runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
-    - name: Setup Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v2
       with:
-        node-version: ${{ matrix.node-version }}
-        cache: 'npm'
+        # Disabling shallow clone is recommended for improving relevancy of reporting
+        fetch-depth: 0
+      # Triggering SonarQube analysis as results of it are required by Quality Gate check
+    - name: SonarQube Scan
+      uses: sonarsource/sonarqube-scan-action@master
+      env:
+        SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        SONAR_HOST_URL: ${{ secrets.SONAR_HOST_URL }}
+    - name: SonarQube Quality Gate check
+      uses: sonarsource/sonarqube-quality-gate-action@master
+      with:
+       scanMetadataReportFile: target/sonar/report-task.txt
+      # Force to fail step after specific time
+      timeout-minutes: 5
+      env:
+       SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+  upload_to_artifactory:
+    name: Artifactory
+    needs: sonar_scan
+    runs-on: ubuntu-latest
+    steps:
+      - name: Archive staging artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: dist-without-markdown
+          path: |
+            dist
+            !dist/**/*.md
+      - name: Archive code coverage results
+        uses: actions/upload-artifact@v2
+        with:
+          name: code-coverage-report
+          path: output/test/code-coverage.html
+  publish_to_staging:
+    name: Publish to staging
+    runs-on: ubuntu-latest
+    needs:
+      - upload_to_artifactory    
+    if: github.event.action == 'Staging'    
+    steps:  
+    - uses: actions/checkout@v2
+    - uses: actions/setup-node@v2.4.0
+      with:
+        registry-url: https://registry.npmjs.org/
+        node-version: 15
     - run: npm ci
-    - run: bash checkmycode.sh
+    - run: npm publish --access restricted
+      env:
+        NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+  publish_to_production:
+    name: Publish to production    
+    runs-on: ubuntu-latest
+    needs:
+      - upload_to_artifactory          
+    if: github.event.action == 'Production'        
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2.4.0
+        with:
+          registry-url: https://registry.npmjs.org/
+          node-version: 15
+      - run: npm ci
+      - run: npm publish --access public
```

Comment:
This is a server. Why publish to npm registry?

Reply:
Accepted, have removed from the currently git actions workflow.